src/sys/ptrace.rs

```diff
@@ -140,3 +140,51 @@ pub fn setsiginfo(pid: Pid, sig: &siginfo_t) -> Result<()> {
         Err(e) => Err(e),
     }
 }
+
+/// Sets the process as traceable with `PTRACE_TRACEME`
+pub fn traceme() -> Result<()> {
+    unsafe {
+        ptrace(
+            ptrace::PTRACE_TRACEME,
+            Pid::from_raw(0),
+            ptr::null_mut(),
+            ptr::null_mut(),
+        ).map(|_| ()) // ignore the useless return value
+    }
+}
+
+/// Makes the `PTRACE_SYSCALL` request to ptrace
```

Comment:
Not the whole thing, but providing a little more detail would be nice. `nix` documentation can also be used as a form of discovery for users who aren't familiar with the POSIX APIs. While I don't expect `nix` to replace the man pages or other sources of documentation, it should allow people unfamiliar with the concepts to understand a little of the functionality.